src/webview/VSCodeKaotoEditorChannelApi.ts

```diff
@@ -95,10 +95,19 @@ export class VSCodeKaotoEditorChannelApi extends DefaultVsCodeKieEditorChannelAp
       const targetFile = path.resolve(classpathRoot, relativePath);
       return new TextDecoder().decode(await vscode.workspace.fs.readFile(vscode.Uri.file(targetFile)));
     } catch (ex) {
-      const errorMessage= `Cannot retrieve content of ${relativePath} relatively to ${classpathRoot}`;
-      vscode.window.showErrorMessage(errorMessage);
-      logInKaotoOutputChannel(errorMessage, ex);
-      return undefined;
+      // TODO: this is an ugly workaround that could lead to path clashes. 2 different APis must be provided to retrieve content, one relative to classpath and the other relative to .kaoto metadata file
+      let kaotoMetadataFile = await this.findExistingKaotoMetadataFile(this.currentEditedDocument.uri);
+      try {
+        if (kaotoMetadataFile !== undefined) {
+          const targetFile = path.resolve(path.dirname(kaotoMetadataFile?.fsPath), relativePath);
```

Comment:
Since the `kaotoMetadataFile` is validated in the `if`statement, I think we can spare the optional chaining.
```suggestion
          const targetFile = path.resolve(path.dirname(kaotoMetadataFile.fsPath), relativePath);
```

Other than that, wouldn't be interesting to also log this exception? Like an `info` log of some sort explaining that we tried to find the file relative to something and it wasn't found, so now we're falling back to relative to the `.kaoto` file.